### PR TITLE
Add nano pytorch example

### PR DESCRIPTION
**Pull request by y199387**
## Description

Add Nano PyTorch examples
- Nano PyTorch IPEX training example.
- Nano PyTorch multi-instance training example.
- Nano PyTorch cv data pipeline example.

### 1. Why the change?
 Add more examples to demonstrate the usage of Nano.

### 2. User API changes
 N/A

### 3. Summary of the change 
 N/A

### 4. How to test?
- [x] N/A

### CODE REVIEW

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                output = model(data)
+                loss = loss_fuc(output, target)
+                self.backward(loss)
+                optimizer.step()
+
+                train_loss += loss.sum()
+                num += 1
+            print(f'Train Epoch: {epoch}, avg_loss: {train_loss / num}')
+
+        # TESTING LOOP
```
Comment (TheaperDeng) [inline python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:109]: are we adapting some external code to this tutorial? if not, we may ignore the testing loop here.

Comment (TheaperDeng) [reply python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:109]: same to other examples

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+import torch
+from torch import nn
+from torch.utils.data import DataLoader
+from torchvision.models import resnet18
+from torchmetrics import Accuracy
+
+from bigdl.nano.pytorch.torch_nano import TorchNano
+
+
```
Comment (TheaperDeng) [inline python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:24]: Since we have some places need to be changed in original training loop, we may add some comments in this example to make them clearer.

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    train_dataset = torch.utils.data.Subset(train_dataset, indices[:-val_size])
+    val_dataset = torch.utils.data.Subset(val_dataset, indices[-val_size:])
+
+    # prepare data loaders
+    train_dataloader = DataLoader(train_dataset, batch_size=32)
+    val_dataloader = DataLoader(val_dataset, batch_size=32)
+
+    return train_dataloader, val_dataloader
+
+
```
Comment (TheaperDeng) [inline python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:78]: `# define a class MyNano derived from our TorchNano`

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        for epoch in range(num_epochs):
+
+            # TRAINING LOOP
+            model.train()
+            train_loss, num = 0, 0
+            for data, target in train_loader:
+                optimizer.zero_grad()
+                output = model(data)
+                loss = loss_fuc(output, target)
+                self.backward(loss)
```
Comment (TheaperDeng) [inline python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:102]: `# replace the loss.backward() with self.backward(loss)`

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Comment by TheaperDeng** [comment]
@MeouSker77 pls have a look either.

File: python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            print(f'Train Epoch: {epoch}, loss: {train_loss/num}')
+
+
+if __name__ == '__main__':
+    # Multi-instance Training
+    #
+    # It is often beneficial to use multiple instances
+    # for training if a server contains multiple sockets or
+    # many cores, so that the workload can make full use of
+    # all CPU cores.
```
Comment (MeouSker77) [inline python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py:112]: also need to explain learning rate scaling & warmup? <https://github.com/intel-analytics/BigDL/pull/5463#discussion_r963776002>

Comment (y199387) [reply python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py:112]: Done

**Review by MeouSker77 — APPROVED** [review]
LGTM

**Comment by TheaperDeng** [comment]
remove `.keep` file and we can merge this PR.

**Comment by y199387** [comment]
> remove `.keep` file and we can merge this PR.

Done

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from torch import nn
+from torch.utils.data import DataLoader
+from torchvision.models import resnet18
+from torchmetrics import Accuracy
+
+from bigdl.nano.pytorch.torch_nano import TorchNano
```
Comment (jason-dai) [inline python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:22]: change to `from bigdl.nano.pytorch import TorchNano` @MeouSker77 @Oscilloscope98

File: python/nano/tutorial/training/pytorch/pytorch_train_ipex.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+import torch
+from torch import nn
+from torch.utils.data import DataLoader
+from torchvision import transforms
+from torchvision.models import resnet18
+from torchvision.datasets import OxfordIIITPet
+from torchmetrics import Accuracy
+
+from bigdl.nano.pytorch.torch_nano import TorchNano
```
Comment (jason-dai) [inline python/nano/tutorial/training/pytorch/pytorch_train_ipex.py:24]: change to `from bigdl.nano.pytorch import TorchNano` @MeouSker77 @Oscilloscope98

File: python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+from torch import nn
+from torch.utils.data import DataLoader
+import torch.nn.functional as F
+from torchvision import transforms
+from torchvision.models import resnet18
+from torchvision.datasets import OxfordIIITPet
+from torchmetrics import Accuracy
+
+from pytorch_lightning import seed_everything
+from bigdl.nano.pytorch.torch_nano import TorchNano
```
Comment (jason-dai) [inline python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py:26]: change to `from bigdl.nano.pytorch import TorchNano` @MeouSker77 @Oscilloscope98

File: python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py (hunk 6 above)
Comment (Oscilloscope98) [reply python/nano/tutorial/training/pytorch/pytorch_cv_data_pipeline.py:22]: Thanks for reminding! Will change it for related how-to guides.
